dhall-json: lower bound too generous

`dhall-json-1.7.5` has changes to support `dhall-1.38.0`. `dhall-1.38.0` introduces [breaking changes](https://hackage.haskell.org/package/dhall-1.38.0/changelog) but `dhall-json` has not raised the lower bound.

Reproduction: `cabal v2-repl -b 'dhall == 1.36.0' -b 'dhall-json == 1.7.5' -j`

Snippet:
```
src/Dhall/JSON.hs:503:9: error:
    • The constructor ‘Core.Lam’ should have 2 arguments, but has been given 3
```

Suggest: raise lower bound on `dhall-json` and publish metadata revision to Hackage.

--- a/dhall-docs/dhall-docs.cabal
+++ b/dhall-docs/dhall-docs.cabal
@@ -65,7 +65,7 @@ Library
         containers                                 ,
         cryptonite                           < 0.29,
         directory            >= 1.3.0.0   && < 1.4 ,
-        dhall                >= 1.35.0    && < 1.39,
+        dhall                >= 1.38.0    && < 1.39,
         file-embed           >= 0.0.10.0           ,
         filepath             >= 1.4       && < 1.5 ,
         lens-family-core     >= 1.0.0     && < 2.2 ,

--- a/dhall-json/dhall-json.cabal
+++ b/dhall-json/dhall-json.cabal
@@ -44,7 +44,7 @@ Library
         aeson-yaml                >= 1.1.0     && < 1.2 ,
         bytestring                                < 0.11,
         containers                >= 0.5.9     && < 0.7 ,
-        dhall                     >= 1.36.0    && < 1.39,
+        dhall                     >= 1.38.0    && < 1.39,
         exceptions                >= 0.8.3     && < 0.11,
         filepath                                  < 1.5 ,
         lens-family-core          >= 1.0.0     && < 2.2 ,

--- a/dhall-lsp-server/dhall-lsp-server.cabal
+++ b/dhall-lsp-server/dhall-lsp-server.cabal
@@ -50,7 +50,7 @@ library
     , containers           >= 0.5.11.0 && < 0.7
     , data-default         >= 0.7.1.1  && < 0.8
     , directory            >= 1.2.2.0  && < 1.4
-    , dhall                >= 1.35.0   && < 1.39
+    , dhall                >= 1.38.0   && < 1.39
     , dhall-json           >= 1.4      && < 1.8
     , filepath             >= 1.4.2    && < 1.5
     , haskell-lsp          >= 0.19.0.0 && < 0.25

--- a/dhall-nix/dhall-nix.cabal
+++ b/dhall-nix/dhall-nix.cabal
@@ -30,7 +30,7 @@ Library
         base                      >= 4.11.0.0 && < 5   ,
         containers                               < 0.7 ,
         data-fix                                 < 0.4 ,
-        dhall                     >= 1.36     && < 1.39,
+        dhall                     >= 1.38     && < 1.39,
         hnix                      >= 0.7      && < 0.13,
         lens-family-core          >= 1.0.0    && < 2.2 ,
         neat-interpolation                       < 0.6 ,

--- a/dhall-openapi/dhall-openapi.cabal
+++ b/dhall-openapi/dhall-openapi.cabal
@@ -49,7 +49,7 @@ library
     base                    >= 4.11.0.0  && < 5    ,
     aeson                   >= 1.0.0.0   && < 1.6  ,
     containers              >= 0.5.8.0   && < 0.7  ,
-    dhall                   >= 1.35.0    && < 1.39 ,
+    dhall                   >= 1.38.0    && < 1.39 ,
     prettyprinter           >= 1.7.0     && < 1.8  ,
     scientific              >= 0.3.0.0   && < 0.4  ,
     sort                    >= 1.0       && < 1.1  ,

